command/cp.go

```diff
@@ -585,15 +604,20 @@ func prepareLocalDestination(
 
 	if isBatch && !flatten {
 		dsturl = dsturl.Join(objname)
-		if err := os.MkdirAll(dsturl.Dir(), os.ModePerm); err != nil {
+		if dryRun {
+			// dryRun => no side effects.
+		} else if err := os.MkdirAll(dsturl.Dir(), os.ModePerm); err != nil {
 			return nil, err
 		}
 	}
 
 	if err == storage.ErrGivenObjectNotFound {
-		if err := os.MkdirAll(dsturl.Dir(), os.ModePerm); err != nil {
+		if dryRun {
+			// dryRun => no side effects.
+		} else if err := os.MkdirAll(dsturl.Dir(), os.ModePerm); err != nil {
```

Comment:
why not `if !dryRun`?

Reply:
@igungor yes, https://github.com/peak/s5cmd/pull/200#discussion_r469279834 is addressed

Reply:
> why not `if !dryRun`?

did, that to avoid having nested if statements